pkgs/os-specific/linux/kernel/common-config.nix

```diff
@@ -590,6 +590,23 @@ let
       CRYPTO_DRBG_HASH                 = yes;
       CRYPTO_DRBG_CTR                  = yes;
 
+      # Set ARCH_MMAP_RND_{COMPAT_,}BITS to the maximum to regain ASLR entropy
+      # https://zolutal.github.io/aslrnt/
+      # https://git.launchpad.net/~ubuntu-kernel/ubuntu/+source/linux/+git/noble/commit/?h=master-next&id=760c2b1fa1f5e95be1117bc7b80afb8441d4b002
+      ARCH_MMAP_RND_BITS = with stdenv.hostPlatform; mkMerge [
```

Comment:
Can you set the mitigation conditionally based on the kernel version?

Reply:
Also... wasn't this fixed in recent kernels?

Reply:
There's https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=4ef9ad19e17676.  I'm not sure about the 64-bit entropy reduction.

Reply:
> Can you set the mitigation conditionally based on the kernel version?
Fixed.

> Also... wasn't this fixed in recent kernels?
https://github.com/torvalds/linux/commit/4ef9ad19e17676b9ef071309bc62020e2373705d
Oops, is it this patch?

Reply:
it looks like next stable kernels are also getting this patch

https://lore.kernel.org/all/ZamGsyJ5JTQ9GYNL@casper.infradead.org/T/#mf00cba2a41cb6a911b49dff4afa125ac6194b120

so 32-bit libraries won't be affected by this now.

64bit ASLR entropy are still reduced by 9 bits, though

Reply:
removed 32bit platforms

Reply:
I wonder if we want to ask some kernel folks what's going on here and what we should be doing.  #linux-hardening on Libera might be a good place.

Reply:
AFAIU the increase by 4bit for x86_64-linux won't account for the entire loss, correct? That's how I understand the article at least:

> It won’t address partial overwites becoming more deterministic though [...]

Anyways, did someone ask upstream for feedback already?

Reply:
Don't think so.